xarray/core/dataset.py

```diff
@@ -2741,6 +2741,63 @@ def roll(self, **shifts):
 
         return self._replace_vars_and_dims(variables)
 
+    def sortby(self, variables, ascending=True):
+        """
+        Sort object by labels or values (along an axis).
+
+        Sorts the dataset, either along specified dimensions,
+        or according to values of 1-D dataarrays that share dimension
+        with calling object.
+
+        If the input variables are dataarrays, then the dataarrays are aligned
+        (via left-join) to the calling object prior to sorting by cell values.
+        NaNs are sorted to the end, following Numpy convention.
+
+        If multiple sorts along the same dimension is
+        given, numpy's lexsort is performed along that dimension:
+        https://docs.scipy.org/doc/numpy/reference/generated/numpy.lexsort.html
+        and the FIRST key in the sequence is used as the primary sort key,
+        followed by the 2nd key, etc.
+
+        Parameters
+        ----------
+        variables: (str, DataArray, or list of either)
+            Name(s) of a 1D variable in coords/data_vars whose values are used
+            to sort the dataset.
+        ascending: boolean, optional.
+            whether to sort by ascending order.
+
+        Returns
+        -------
+        sorted: Dataset
+            A new dataset where all the specified dims are sorted by dim
+            labels.
+        """
+        from .dataarray import DataArray
+
+        if not isinstance(variables, list):
+            variables = [variables]
+        else:
+            variables = variables
+        variables = [v if isinstance(v, DataArray)
```

Comment:
nit: I prefer writing this like:
```
        variables = [v if isinstance(v, DataArray) else self[v]
                     for v in variables]
```
because the `else` clause is part of the `if` conditional statement.